pkg/sql/job_exec_context.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package sql
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/lease"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sessiondata"
+)
+
+// plannerJobExecContext is a wrapper to implement JobExecContext with a planner
+// without allowing casting directly to a planner. Eventually it would be nice
+// if we could implement the API entirely without a planner however the only
+// implementation of extendedEvalContext is very tied to a planner.
```

Comment:
Would this also allow us to life job exec context out of the sql package into jobs? That would also unlock the ability to actually pass in a job exec context in the resumer rather than an `interface{}`, which would also be nice.